dask/dataframe/io.py

```diff
@@ -637,26 +637,54 @@ def to_castra(df, fn=None, categories=None, sorted_index_column=None,
         return delayed([Delayed(key, [dsk]) for key in keys])
 
 
-def to_csv(df, filename, compression=None, get=None, compute=True, **kwargs):
+def to_csv(df, filename, name_function=None, compression=None, get=None, compute=True, **kwargs):
     if compression:
         raise NotImplementedError("Writing compressed csv files not supported")
     name = 'to-csv-' + uuid.uuid1().hex
 
+    kwargs2 = kwargs.copy()
+
+    if name_function is None:
+        name_function = build_name_function(df.npartitions - 1)
+
+    if '*' in filename:
+        formatted_names = [name_function(i) for i in range(df.npartitions)]
+        if formatted_names != sorted(formatted_names):
+            warn("In order to preserve order between partitions "
+                 "name_function must preserve the order of its input")
+
+        if filename.count('*') > 1:
+            raise ValueError("A maximum of one asterisk is accepted in filename")
+
+        if 'mode' in kwargs and kwargs['mode'] != 'w':
+            raise ValueError("to_csv does not support writing to multiple files in append mode, "
+                             "please specify mode='w'")
+
+        link = False
+    else:
+        kwargs2['mode'] = 'a'
+        kwargs2['header'] = False
+        link = True
```

Comment:
I suggest that we keep `link` as it is now for consistency and then change it throughout the codebase in a future PR.  

I agree with @dukebody that while the term `link` makes a lot of sense to us as developers it's probably not clear from a naive user perspective.